src/text-editor.coffee

```diff
@@ -3192,11 +3187,40 @@ class TextEditor extends Model
     Grim.deprecate("This is now a view method. Call TextEditorElement::getWidth instead.")
     @displayBuffer.getWidth()
 
-  getScrollRow: -> @scrollRow
-  setScrollRow: (@scrollRow) ->
+  # Experimental: Scroll the editor such that the given screen row is at the
+  # top of the visible area.
+  setFirstVisibleScreenRow: (screenRow, fromView) ->
```

Comment:
This may need `NaN` detection. Otherwise we get a nasty `displayBuffer` error. May require closing and re-opening tab after calling `editor.setFirstVisibleScreenRow(NaN)`.